### ISSUE TITLE
Error uploading artifact to container registry

### ISSUE THREAD

**Issue by qrkourier** [actions/attest-build-provenance#68]
I followed [this guide](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images) to add a workflow for publishing an image with an attestation.

The build-push step successfully pushed to [this Hub repo](https://hub.docker.com/r/kbinghamnetfoundry/ziti-console-assets/tags) with this workflow. It's failing at the attestation step with "Error uploading artifact to container registry."

```
Run actions/attest-build-provenance@v1
Run actions/attest-build-provenance/predicate@db1dde0f270afe12073070ac7aa802958ae3ec04
  
Run actions/attest@d442d85e120905fa5821e76edb1e07a75b94ee7a
  
Attestation created for docker.io/***/ziti-console-assets@sha256:fbc45cffeb4effc49260f09f1a4e12041320fdb0e68f34f6d3f4ef508216bdf2
Attestation signed using certificate from Public Good Sigstore instance
Attestation signature uploaded to Rekor transparency log
https://search.sigstore.dev?logIndex=92703814
Attestation uploaded to repository
https://github.com/qrkourier/ziti-console/attestations/822597
Error: Error uploading artifact to container registry
```

build log from run: https://github.com/qrkourier/ziti-console/actions/runs/9035775334/job/24831227577

Here's my job step.

```yaml
jobs:
  docker-build-push:
    steps:
# ...snip...
        - name: Generate artifact attestation
          uses: actions/attest-build-provenance@v1
          with:
            subject-name: docker.io/kbinghamnetfoundry/ziti-console-assets
            subject-digest: ${{ steps.push.outputs.digest }}
            push-to-registry: true
```

Thank you for your help.

**Comment by qrkourier** [comment]
I get a slightly different error when I switch the source of the Docker Hub login username from a GHA secret to a var so that it's unmasked.

> Error: OCIError: Error uploading artifact to container registry
FetchError: network timeout at: https://www.docker.com/


```
Run actions/attest-build-provenance@v1
Run actions/attest-build-provenance/predicate@db1dde0f270afe12073070ac7aa802958ae3ec04
  
Run actions/attest@32f49af6653ef9b7d1a40182d53fc9ebd53447d8
  
Attestation created for docker.io/kbinghamnetfoundry/ziti-console-assets@sha256:f169fe1ec4ca09dbeebe15658c607ac0244c6861147c81bfaca4d4f776063634
Attestation signed using certificate from Public Good Sigstore instance
Attestation signature uploaded to Rekor transparency log
https://search.sigstore.dev?logIndex=92721576
Attestation uploaded to repository
https://github.com/qrkourier/ziti-console/attestations/822796
Error: OCIError: Error uploading artifact to container registry
FetchError: network timeout at: https://www.docker.com/
```

from: https://github.com/qrkourier/ziti-console/actions/runs/9036298596/job/24833209564

**Comment by bdehamer** [comment]
@qrkourier When setting `subject-name` try using `index.docker.io` as the image registry name instead of `docker.io`:

```yaml
  - name: Generate artifact attestation
    uses: actions/attest-build-provenance@v1
    with:
      subject-name: index.docker.io/kbinghamnetfoundry/ziti-console-assets
      subject-digest: ${{ steps.push.outputs.digest }}
      push-to-registry: true
```

**Comment by qrkourier** [comment]
That worked. I can use the same, predictable string `index.docker.io` as an env var like `REGISTRY` throughout the workflow, except the build-push-action requires `docker.io`, apparently. I got a 401 pushing to the index subdomain.

**Comment by qrkourier** [comment]
I meant to clarify that `index.docker.io` is valid for login and attestation, but not push.

**Comment by bdehamer** [comment]
> I meant to clarify that index.docker.io is valid for login and attestation, but not push.

I'm not sure what may be going on there. I've definitely used `index.docker.io` in conjunction with the `build-push-action`:

https://github.com/bdehamer/attest-demo/blob/main/.github/workflows/oci-docker-hub.yml#L34-L40

**Comment by qrkourier** [comment]
Interesting. I was, admittedly, leaping to that conclusion on the basis of a single attempt.

> Error: buildx failed with: ERROR: failed to solve: failed to push index.docker.io/kbinghamnetfoundry/ziti-console-assets:main: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://auth.docker.io/token?scope=repository%3Akbinghamnetfoundry%2Fziti-console-assets%3Apull%2Cpush&service=registry.docker.io: 401 Unauthorized

from: https://github.com/qrkourier/ziti-console/actions/runs/9036612271/job/24833832246

This was one of several runs wherein I fiddled the registry domain name while taking your advice to try the "index" variant.  In this attempt, I tried to use `index.docker.io` as a drop-in replacement everywhere I had been use `docker.io`.

**Comment by bdehamer** [comment]
The situation around what the correct host prefix should be for Docker Hub hosted images is pretty [confusing](https://github.com/docker/hub-feedback/issues/2113). I think the right answer here is that the action should gracefully handle all the different variants so that you don't have to worry about it. Will see about getting a fix in for this.

**Comment by bdehamer** [comment]
Just published [v1.1.2](https://github.com/actions/attest-build-provenance/releases/tag/v1.1.2) of the action which should allow you to use `docker.io` as the registry domain when interacting with the Docker Hub.
